File: 2주차/0612/지은/k진수에서 소수 개수 구하기.py
def is_prime(nu):   #소수 판별
    if nu==1:
        return False
    for i in range(2, int(nu**0.5)+1):
        if nu%i==0:
            return False
    return True

def solution(n, k):
    answer = 0
    n_k = ''    #k진수로 변환한 n

    #정수 n을 k진수로 바꾼다
    while n:
        n, m = n//k, n%k
        n_k += str(m)
    n_k = n_k[::-1]
    
    nums = n_k.split('0')   #0을 기준으로 split

    for num in nums:
        if num=='':
            continue
        if is_prime(int(num)):
            answer += 1

    return answer

File: 2주차/0612/찬영/k 진수에서 소수 개수 구하기.java
import java.util.ArrayList;
import java.util.Arrays;

class Solution {

	public int solution(int n, int k) {
		String primeNumber = "";
		
		while(true) {
			primeNumber = Integer.toString(n%k) + primeNumber;

			n /= k;
			if (n == 0) {
				break;
			}
		}
		
		ArrayList<String> list = new ArrayList<String>();
		for (int i = 0; i<primeNumber.length(); i++) {
			list.add(String.valueOf(primeNumber.charAt(i)));
		}
		
		ArrayList<String> number = new ArrayList<String>();
		String num = "";
		for(String x : list) {
			if(x.equals("0")) {
				if(!num.equals("")) {
					number.add(num);
					num = "";
				}
			}else {
				num += x;
			}
		}
		if (!num.equals("")) {
			number.add(num);
		}

	
		int cnt = 0;
		for(String x : number) {

			if (isPrime(Long.parseLong(x))) {
				cnt += 1;
			}
		}
		return cnt;
		
	}
	public boolean isPrime(Long n) {
		if (n == 1) {
			return false;
		}
		int number = (int) Math.sqrt(n);
		for (int i = 2; i < number+1; i++) {
			if (n % i ==0) {
				return false;
			}
		}
		return true;
	}
}
